cosmos/lib/cosmos/utilities/s3_file_cache.rb

```diff
@@ -178,7 +183,11 @@ def initialize(name = 'default', max_disk_usage = MAX_DISK_USAGE)
         file = @cached_files.get_next_to_retrieve
         # Cosmos::Logger.debug "Next file: #{file}"
         if file and (file.size + @cached_files.current_disk_usage()) <= @max_disk_usage
-          file.retrieve
+          begin
+            file.retrieve
+          rescue
+            # Will be automatically retried
+          end
```

Comment:
What's the point of raising the error and then silently rescuing and retrying?

Reply:
The error didn't use to be there, but I added it for when using retrieve by itself